README.md

```diff
@@ -1,137 +1,326 @@
 # generate_parameter_library
-This package aims to automate parameter handling within ROS 2. The build script automatically generates a c++ library from a yaml file input. The generated library contains a struct with all parameters as nested fields. Additional methods are generated, which enable parameter updating.
 
-# Usage
-You must include the generate_parameter_library package in your cmake project.
-
-`find_package(generate_parameter_library REQUIRED)`
-
-You then need call the `generate_parameter_library` function with the following arguments:
-```
-LIB_NAME # cmake target name for the parameter library
-YAML_FILE # path to input yaml file
-[VALIDATE_HEADER] # Optional path to header containing custom validators
-```
-
-See [example project](example/CMakeLists.txt) to understand how to use this.
-
-# Example:
-## Build the node
-```
- mkdir colcon_ws
- mkdir colcon_ws/src
- cd colcon_ws/src
- git clone https://github.com/picknikrobotics/generate_parameter_library.git
- cd ..
- colcon build
+This package aims to automate parameter handling within ROS 2 via c++ code generation. A user can specify parameter
+names, types, descriptions, and validation functions all within the YAML syntax. This packages provides a
+straightforward interface to integrate the code generation into cmake builds. The generated library contains a c++
+struct with all specified parameters as nested fields. Additionally, methods are generated, which enable dynamic parameter
+updating, parameter copying, and parameter validation.
+
+## Generated code interface
+The generated code is primarily consists of two major component: 1) a pure data struct `struct Params` that contains values of all parameters and 2) a class `ParamListener` that handles parameter declaration, updating, and validation. The general structure is shown below.
```

Comment:
```suggestion
This package aims to automate parameter handling within ROS 2 via c++ code generation.
A user can specify parameter names, types, descriptions, and validation functions all within the YAML syntax.
This packages provides a straightforward interface to integrate the code generation into cmake builds.
The generated library contains a c++ struct with all specified parameters as nested fields. Additionally, methods are generated, which enable dynamic parameter updating, parameter copying, and parameter validation.

## Generated code interface
The generated code is primarily consists of two major components:
1) a pure data struct `struct Params` that contains values of all parameters and
2) a class `ParamListener` that handles parameter declaration, updating, and validation.
The general structure is shown below.
```

In markdown you should use semantic newlines where each sentence is a separate line.